app/src/test/java/cm/aptoide/pt/app/AppViewManagerTest.java

```diff
@@ -0,0 +1,695 @@
+import cm.aptoide.pt.appview.PreferencesManager;
+import cm.aptoide.pt.database.realm.Download;
+import cm.aptoide.pt.database.realm.MinimalAd;
+import cm.aptoide.pt.dataprovider.model.v7.store.Store;
+import cm.aptoide.pt.dataprovider.ws.v2.GenericResponseV2;
+import cm.aptoide.pt.download.AppContext;
+import cm.aptoide.pt.download.DownloadFactory;
+import cm.aptoide.pt.download.InstallType;
+import cm.aptoide.pt.download.Origin;
+import cm.aptoide.pt.install.Install;
+import cm.aptoide.pt.install.InstallAnalytics;
+import cm.aptoide.pt.install.InstallManager;
+import cm.aptoide.pt.notification.NotificationAnalytics;
+import cm.aptoide.pt.search.model.SearchAdResult;
+import cm.aptoide.pt.store.StoreUtilsProxy;
+import cm.aptoide.pt.timeline.SocialRepository;
+import cm.aptoide.pt.view.AppViewConfiguration;
+import cm.aptoide.pt.view.app.AppCenter;
+import cm.aptoide.pt.view.app.AppStats;
+import cm.aptoide.pt.view.app.AppsList;
+import cm.aptoide.pt.view.app.DetailedApp;
+import cm.aptoide.pt.view.app.DetailedAppRequestResult;
+import cm.aptoide.pt.view.app.FlagsVote;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import rx.Completable;
+import rx.Observable;
+import rx.Single;
+
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by D01 on 28/05/2018.
+ */
+
+public class AppViewManagerTest {
+
+  private final int limit = 18;
+  @Mock private InstallManager installManager;
+  @Mock private DownloadFactory downloadFactory;
+  @Mock private AppCenter appCenter;
+  @Mock private ReviewsManager reviewsManager;
+  @Mock private AdsManager adsManager;
+  @Mock private StoreManager storeManager;
+  @Mock private FlagManager flagManager;
+  @Mock private StoreUtilsProxy storeUtilsProxy;
+  @Mock private AptoideAccountManager aptoideAccountManager;
+  @Mock private AppViewConfiguration appViewConfiguration;
+  @Mock private InstallAnalytics installAnalytics;
+  @Mock private PreferencesManager preferencesManager;
+  @Mock private DownloadStateParser downloadStateParser;
+  @Mock private AppViewAnalytics appViewAnalytics;
+  @Mock private NotificationAnalytics notificationAnalytics;
+  @Mock private DetailedApp cachedApp;
+  @Mock private MinimalAd minimalAd;
+  @Mock private DetailedAppRequestResult detailedAppRequestResult;
+  @Mock private ReviewRequestResult reviewRequestResult;
+  @Mock private MinimalAdRequestResult minimalAdRequestResult;
+  @Mock private AppsList appsList;
+  @Mock private SearchAdResult searchAdResult;
+  @Mock private Store store;
+  @Mock private AppStats stats;
+  @Mock private SocialRepository socialRepository;
+  @Mock private GenericResponseV2 genericResponseV2;
```

Comment:
Does this have to be a mock? Personally i would have just created it on the setup method.

Reply:
After discussing this, we agreed that we should not be using this GenericResponseV2 object in the FlagManager, so we will have to change this in the future. So for now, we should be mocking this response object.